src/components/graph/search/GraphSearchConfirm.tsx

```diff
@@ -0,0 +1,77 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ***************************************************************************** */
+
+import React from 'react';
+import KeyCodes from '../../../util/KeyCodes';
+
+interface Props {
+	onConfirm: () => void;
+	onDecline: () => void;
+}
+
+const GraphSearchConfirm = ({ onConfirm, onDecline }: Props) => {
+	const [countdown, setCountdown] = React.useState(5);
+	const timeoutRef = React.useRef<NodeJS.Timeout>();
+	const intervalRef = React.useRef<NodeJS.Timeout>();
+
+	React.useEffect(() => {
+		timeoutRef.current = setTimeout(onDecline, 5000);
+		intervalRef.current = setInterval(() => {
+			setCountdown(c => c - 1);
+		}, 1000);
+		return () => {
+			if (timeoutRef.current) {
+				clearTimeout(timeoutRef.current);
+			}
+			if (intervalRef.current) {
+				setCountdown(0);
+				clearInterval(intervalRef.current);
+			}
+		};
+	}, []);
+
+	React.useEffect(() => {
+		const listener = (e: KeyboardEvent) => {
+			if (e.keyCode === KeyCodes.ENTER) {
+				onConfirm();
+				return;
+			}
+			if (e.keyCode === KeyCodes.ESCAPE) {
+				onDecline();
+			}
+		};
+		document.addEventListener('keydown', listener);
+		return () => {
+			document.removeEventListener('keydown', listener);
+		};
+	});
```

Comment:
there should be empty dependency list otherwise you will add delete keydown listeners everytime countdown changes or component updates